server/s3_put_multiobject_action.cc

```diff
@@ -134,6 +134,12 @@ void S3PutMultiObjectAction::check_part_details() {
   } else if ((request->get_object_name()).length() > MAX_OBJECT_KEY_LENGTH) {
     set_s3_error("KeyTooLongError");
     send_response_to_s3_client();
+  } else if (request->get_data_length() <= MINIMUM_ALLOWED_PART_SIZE) {
```

Comment:
This check for MINIMUM_ALLOWED_PART_SIZE can  be a problem if the part happens to be the last one
last part can be of size less than MINIMUM_ALLOWED_PART_SIZE